### PR TITLE
First draft 1501 course guide

### PR DESCRIPTION
**Pull request by NathanBarta**

### CODE REVIEW

File: src/guides/academics/succeed-in-course-x/succeed-in-1501.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+4. Build the project incrementally. Develop unit tests as you go along, it will be difficult to retroactively build them. Trace though your code when you make additions/changes - make sure everything is still working as expected. If your professor allows multiple submissions, submit frequently to gain partial credit in the event you are not able to turn in a completed project.
+5. (2-3 days before deadline) Build tests for your complete project. Look for edge cases in your code. Consider [fuzzing](https://en.wikipedia.org/wiki/Fuzzing) as it's a relatively easy way to find bugs and make sure your testing coverage is adequate. 
+
+You do not want to be holding on to an incomplete project the last 2-3 days before it's due. That's likely when the next project is being introduced - which means you'll have extra things to think about. Also, TA office hours (which you should be going to if you need help) get really busy towards the deadline, so getting help may be a challenge.
+
+> Don't cheat. It's very easy to catch. Beyond the typical meaning of cheating: *don't cheat yourself*. Did you write some code that you don't fully understand? Figure it out. Do it right.
```
Comment (agrattan0820) [inline src/guides/academics/succeed-in-course-x/succeed-in-1501.md:166]: `It's very easy to catch.` to `It's very easy to get caught."?

**Review by agrattan0820 — APPROVED** [review]
Awesome

File: src/guides/academics/succeed-in-course-x/succeed-in-1501.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+  - Approaches
+    - Supervised learning
+    - Unsupervised learning
+    - Semi-supervised learning
+    - Reenforcement learning
+  - Clustering
+    - K-means
+      - Initial assignment
+      - Lloyd's algorithm
+      - Distance metric
```
Comment (FishOfPitt116) [inline src/guides/academics/succeed-in-course-x/succeed-in-1501.md:105]: With 1503 becoming a requirement, is this still true? Will this be included in 1501? I guess time will tell

File: src/guides/academics/succeed-in-course-x/succeed-in-1501.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+Asking questions is a great way to clarify topics. You are going to have questions at some point. Even if you could search it up online, why would you? You're depriving yourself of a few great benefits: 
+
+1. Vocalizing a question: Communication matters a ton. If you aren't able to communicate with your colleagues / coworkers, you are holding people back. Additionally, if you can't develop internal dialog, how are you going to do all of your thinking? The reading material for your courses is going to get harder and harder, so are your lectures - develop your eyes and ears. 
+2. Assisting your professor: If you aren't understanding something, it could be because your professor omitted some details or got off track. It's not easy to jump between classes and "meet" everybody at the level they are at. Asking a question will help them get back on track - they will be appreciative. At the very least, it will probably help with the pacing of the lecture.
+3. Assisting your classmates: Your classmates probably have similar questions!
+4. Actively thinking of questions is good for your brain. Why do we do things X way? What could happen if we did Y?
+
+Take notes!
+
+### **Recitations**
```
Comment (FishOfPitt116) [inline src/guides/academics/succeed-in-course-x/succeed-in-1501.md:133]: Emphasize that you should attend recitation (despite the fact that I may or may not have throughout the semester I took 1501...)

You could also mention that recitation might be required depending on professor

File: src/guides/academics/succeed-in-course-x/succeed-in-1501.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+### **Readings**
+
+Readings are strongly recommended. The textbook often includes interesting auxiliary information, and thought experiments not present in lecture. You can read at your own pace.
+
+Don't restrict yourself to course textbooks! Think of questions based off of your readings and do some research. It will lead you down some really interesting paths and help you get the most out of the course. As a general rule reading 2-3 articles about a given topic is a good way to work towards building mastery.
+
+Code is meant to be read too!
+
+I'd recommend reading the author's implementations and making sure it lines up with what you read from the book. Once you can draw the connections, move on, or even experiment with alternate implementations. Why did the author do it that way?
```
Comment (FishOfPitt116) [inline src/guides/academics/succeed-in-course-x/succeed-in-1501.md:147]: Mention that most profs have author's code that they provide for you as the semester goes on. Sometimes, in Farnan's project specifically, he lets you directly pull from this code for your data structures. So make a point that this should be something that's referenced

File: src/guides/academics/succeed-in-course-x/succeed-in-1501.md (hunk 1 above)
Comment (FishOfPitt116) [reply src/guides/academics/succeed-in-course-x/succeed-in-1501.md:166]: Also mention using TAs and tutors as a resource. "Figure it out" gives off the impression that they may feel they may be on their own

**Review by FishOfPitt116 — CHANGES_REQUESTED** [review]
Overall, really good article that covers a lot of bases when it comes to knowing what 1501 covers and how to approach assignments. Just make the few minor changes and I'll approve :)

File: src/guides/academics/succeed-in-course-x/succeed-in-1501.md (hunk 2 above)
Comment (NathanBarta) [reply src/guides/academics/succeed-in-course-x/succeed-in-1501.md:105]: Yeah, we'll have to keep our eyes on it.

**Review by FishOfPitt116 — APPROVED** [review]
